File: domusviewer/app/src/main/java/it/achdjian/paolo/domusviewer/DomusEngineRest/GetDevice.java
package it.achdjian.paolo.domusviewer.DomusEngineRest;

import android.content.SharedPreferences;
import android.support.annotation.NonNull;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

/**
 * Created by Paolo Achdjian on 15/04/16.
 */
public class GetDevice  extends DomusEngineRest  {
    private static ObjectMapper MAPPER = new ObjectMapper();
    public interface Listener {
        void newDevice(JSonDevice zDevice);
    }

    private final int nwkAddress;
    @NonNull
    private final Listener listener;

    public GetDevice(@NonNull SharedPreferences sharedPreferences, @NonNull ConnectionStatus connected, int nwkAddress, @NonNull Listener listener) {
        super(sharedPreferences, connected);
        this.nwkAddress = nwkAddress;
        this.listener = listener;
    }

    @Override
    public void run() {
        String body = get("/devices/"+Integer.toString(nwkAddress,16));
        if (body != null) {
            if (!body.isEmpty()) {
                try {
                    JSonDevice zDevice = MAPPER.readValue(body, JSonDevice.class);
                    listener.newDevice(zDevice);
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: temperature_monitor/app/src/main/java/it/achdjian/paolo/temperaturemonitor/domusEngine/rest/GetDevice.kt
package it.achdjian.paolo.temperaturemonitor.domusEngine.rest

import android.util.Log
import it.achdjian.paolo.temperaturemonitor.domusEngine.DomusEngine
import it.achdjian.paolo.temperaturemonitor.domusEngine.MessageType
import java.io.IOException

/**
 * Created by Paolo Achdjian on 7/11/17.
 */
class GetDevice(val nwkAddress: Int, val domusEngineRest: DomusEngineRest, val domusEngine: DomusEngine) : ZigbeeRunnable() {
    override fun run() {
        val body = domusEngineRest.get("/devices/" + nwkAddress.toString(16))
        if (body.isNotBlank()) {
            try {
                Log.i("REST", body)
                val zDevice = MAPPER.readValue(body, JsonDevice::class.java)
                domusEngine.handler.sendMessage(domusEngine.handler.obtainMessage(MessageType.NEW_DEVICE, zDevice))
                Log.i("REST", zDevice.toString())
            } catch (e: IOException) {
                Log.e(TAG, "Error parsing response for /devices/" + nwkAddress)
                Log.e(TAG, "Response: " + body)
                e.printStackTrace()
            }
        }
    }

}